sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -79,26 +80,37 @@ public SentryEvent process(@NotNull SentryEvent event, @NotNull Hint hint) {
 
     // the app start measurement is only sent once and only if the transaction has
     // the app.start span, which is automatically created by the SDK.
-    if (!sentStartMeasurement && hasAppStartSpan(transaction)) {
-      final @NotNull TimeSpan appStartTimeSpan =
-          AppStartMetrics.getInstance().getAppStartTimeSpanWithFallback(options);
-      final long appStartUpDurationMs = appStartTimeSpan.getDurationMs();
-
-      // if appStartUpDurationMs is 0, metrics are not ready to be sent
-      if (appStartUpDurationMs != 0) {
-        final MeasurementValue value =
-            new MeasurementValue(
-                (float) appStartUpDurationMs, MeasurementUnit.Duration.MILLISECOND.apiName());
-
-        final String appStartKey =
-            AppStartMetrics.getInstance().getAppStartType() == AppStartMetrics.AppStartType.COLD
-                ? MeasurementValue.KEY_APP_START_COLD
-                : MeasurementValue.KEY_APP_START_WARM;
-
-        transaction.getMeasurements().put(appStartKey, value);
+    if (hasAppStartSpan(transaction)) {
+      if (!sentStartMeasurement) {
+        final @NotNull TimeSpan appStartTimeSpan =
+            AppStartMetrics.getInstance().getAppStartTimeSpanWithFallback(options);
+        final long appStartUpDurationMs = appStartTimeSpan.getDurationMs();
+
+        // if appStartUpDurationMs is 0, metrics are not ready to be sent
+        if (appStartUpDurationMs != 0) {
+          final MeasurementValue value =
+              new MeasurementValue(
+                  (float) appStartUpDurationMs, MeasurementUnit.Duration.MILLISECOND.apiName());
+
+          final String appStartKey =
+              AppStartMetrics.getInstance().getAppStartType() == AppStartMetrics.AppStartType.COLD
+                  ? MeasurementValue.KEY_APP_START_COLD
+                  : MeasurementValue.KEY_APP_START_WARM;
+
+          transaction.getMeasurements().put(appStartKey, value);
+
+          attachColdAppStartSpans(AppStartMetrics.getInstance(), transaction);
+          sentStartMeasurement = true;
+        }
+      }
 
-        attachColdAppStartSpans(AppStartMetrics.getInstance(), transaction);
-        sentStartMeasurement = true;
+      final @Nullable App appContext = transaction.getContexts().getApp();
+      if (appContext != null) {
```

Comment:
`l`: I know it should be set by the DefaultAndroidEventProcessor which is running before this one, but should we fall back to creating a new instance of `App` in case it's null?